feat: implement a user dictionary to set costs

abetomo opened the pull request:
I want to set the cost in the user dictionary.
I've implemented this in a small diff, but please let me know if there are any project policies or rules for implementation.

abetomo committed:
feat: implement a user dictionary to set costs

abetomo committed:
test: add a test to load user dictionary with cost

abetomo committed:
docs: add about format of the user dictionary

add that the same format as the IPA dictionary can also be used.

johtani commented on lindera-ipadic-builder/src/lib.rs:
```diff
@@ -94,6 +94,9 @@ impl<'a> CSVRow<'a> {
 
     fn from_line_user_dict(line: &'a String) -> CSVRow<'a> {
         let fields: Vec<_> = line.split(",").collect();
+        if fields.len() >= 13 {
```
Great work!
One suggestion/question about the wrong CSV format.

I think we should notify errors if the length of columns is wrong. e.g. 12
Because the user forgot the 13th column, then Lindera creates the CSVRow as a simple user dictionary.
That is unexpected behavior.

日本語でも。
CSVのカラムの数のチェックをしたほうがいいかと思います。
もし、12カラムしかない行があった場合、simple user dictionaryとして扱おうとします。
カラムの情報がずれておかしなことになるかと。

abetomo replied:
確認ありがとうございます。
カラム数をチェックするようにしました。

johtani commented on lindera/src/tokenizer.rs:
```diff
@@ -460,6 +460,44 @@ mod tests {
         );
     }
 
+    #[test]
```
Could you add the mixed CSV pattern, e.g. one CSV has a simple and detailed user dictionary to Unit test?

simpleとdetailedの両方の行をもつCSVのテストケースも追加していただけますか？

abetomo replied:
いくつかテストケースを追加いたしました。

johtani reviewed (changes requested):
Great work!

I left one suggestion, please take a look?

abetomo committed:
test: add a test with a mix of simple and detailed user dictionary

changed the test name and CSV file name to match the type of user dictionary described in the README.

abetomo committed:
fix: fix typo in message

wordost ->
word_cost

abetomo committed:
feat: implement a check for the number of fields in a user dictionary

johtani reviewed (approved):
LGTM!
Thanks for your contribution!
